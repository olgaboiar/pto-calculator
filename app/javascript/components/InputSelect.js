import React from 'react'
import Select from 'react-select'

class InputSelect extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      positionOptions: this.props.options
    };
  
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange = position => {
    this.props.onChange(position);
  }
  
  render() {
    const positionOptions = this.props.options;
    return (
      <div className="form-group row center-align" data-testid="position">
        <div className="col s12 m6 l4 offset-l4">
          <label>Select Your Position: </label>
          <Select
            options={positionOptions}
            onChange={this.handleChange}
            defaultValue={positionOptions[0]}
          />
        </div>
      </div>
    );
  }
}

export default InputSelect;
