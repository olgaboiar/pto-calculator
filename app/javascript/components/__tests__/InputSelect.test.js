import { shallow, mount } from 'enzyme';
import * as React from 'react';
import InputSelect from '../InputSelect'

describe('InputSelect.js', () => {
  const testOptions = [
    { value: 'chocolate', label: 'Chocolate' },
    { value: 'strawberry', label: 'Strawberry' },
    { value: 'vanilla', label: 'Vanilla' }
  ]

  it('renders correctly', () => {
    const wrapper = shallow(
      <InputSelect options={testOptions} />,
    );
  
    expect(wrapper).toMatchSnapshot();
  });

  it('displays a default value', () => {
    const wrapper = mount(
      <InputSelect 
        options={testOptions}
        value={testOptions[1].value}
      />);
    
    expect(wrapper.html()).toContain('Strawberry');
  });
})
