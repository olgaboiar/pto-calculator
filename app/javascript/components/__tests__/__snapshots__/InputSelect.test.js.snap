// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSelect.js renders correctly 1`] = `ShallowWrapper {}`;
